packages/riipen-ui/src/components/Chip.jsx

```diff
@@ -1,330 +1,399 @@
+    { clickable },
+    { focusVisible },
+    hover ? "hover" : null
+  );
 
-    /**
-     * The color to use.
-     */
-    color: PropTypes.oneOf([
-      "default",
-      "dark",
-      "primary",
-      "secondary",
-      "tertiary",
-      "positive",
-      "warning",
-      "negative"
-    ]),
-
-    /**
-     * The component used for the root node.
-     * Either a string to use a DOM element or a component.
-     */
-    component: PropTypes.elementType,
+  const handleBlur = () => {
+    setFocusVisible(false);
+    onBlurVisible();
+  };
 
-    /**
-     * If `true`, the chip should be displayed in a disabled state.
-     */
-    disabled: PropTypes.bool,
+  const handleClick = () => {
+    if (onClick) onClick();
+  };
 
-    /**
-     * Whether the chip should invert its variant on hover.
-     */
-    hover: PropTypes.bool,
+  const handleFocus = e => {
+    setFocusVisible(isFocusVisible(e));
+  };
 
-    /**
-     * Icon to display at start of chip.
-     */
-    icon: PropTypes.node,
+  return (
+    <React.Fragment>
+      <Component
+        className={className}
+        onBlur={handleBlur}
+        onClick={handleClick}
+        onFocus={handleFocus}
+        ref={ref}
+        tabIndex={clickable ? "0" : undefined}
+      >
+        {icon && <span className="icon">{icon}</span>}
+        <span className={clsx("label")}>{label || children}</span>
+      </Component>
+      <style jsx>{`
+        .root {
+          background-color: ${theme.palette.grey[300]};
+          border-radius: 50px;
+          border: 1px solid;
+          border-color: transparent;
+          box-sizing: border-box;
+          color: ${theme.palette.text.secondary};
+          cursor: default;
+          display: inline-flex;
+          font-family: ${theme.typography.fontFamily};
+          font-size: 13px;
+          justify-content: center;
+          outline: 0;
+          vertical-align: middle;
+          white-space: nowrap;
+        }
+        .root.xsmall {
+          padding: 1px 5px;
+        }
+        .root.small {
+          padding: 5px 10px;
+        }
+        .root.medium {
+          padding: 10px 15px;
+        }
+        .root.disabled {
+          opacity: 0.6;
+          pointer-events: none;
+        }
+        .root.clickable {
+          cursor: pointer;
+        }
+        .root.clickable:hover,
+        .root.focusVisible,
+        .root.hover:hover {
+          background-color: transparent;
+          border: 1px solid ${theme.palette.text.primary};
+        }
+        .icon {
+          align-items: center;
+          display: flex;
+          padding-right: 10px;
+          white-space: nowrap;
+        }
 
-    /**
-     * The content of the label.
-     */
-    label: PropTypes.node,
+        .label {
+          align-items: center;
+          display: flex;
+          white-space: nowrap;
+        }
 
-    /**
-     * Action to perform when clicked.
-     */
-    onClick: PropTypes.func,
+        .primary {
```

Comment:
What's our rule on whitespace between css in the ui kit?  Some of these have newlines between them, some don't

Reply:
We actually don't really have a rule. I just try to group them to what makes sense given the context.